Source/Models/BrickDimension.swift

```diff
@@ -18,6 +18,41 @@ public struct BrickSize {
     }
 }
 
+public enum RestrictedSize {
+    case minimumSize(size: BrickDimension)
+    case maximumSize(size: BrickDimension)
+}
+
+public struct RestrictedBrickSize {
```

Comment:
Why is this a separate struct and not just part of the BrickSize `restricted`? I'd had expected a new case in the enum `indirect case restricted(size: BrickDimension, restriction: RestrictedBrickSize)`